web/beacon-app/src/features/projects/routes/ProjectDetailPage.tsx

```diff
@@ -23,25 +23,20 @@ const ProjectDetailPage = () => {
   const { id: projectID } = param;
 
   invariant(projectID, 'project id is required');
-  const { hasProject, hasTopics, hasApiKeys, warningMessage, hasAlreadySetup, hasTenant } =
+  const { hasProject, hasTopics, hasApiKeys, warningMessage, hasAlreadySetup } =
     useProjectSetup(projectID);
-  const { project } = useFetchProject(projectID);
+  const { project, error } = useFetchProject(projectID);
 
   const getNormalizedProjectName = () => {
     return project?.name?.split('-').join(' ');
   };
 
   useEffect(() => {
-    if (!param || !projectID) {
+    // when user switch to another organization and the current project is not found then redirect to projects page
```

Comment:
Thanks for including this fix and the comment so that no one will forget why this was added.